### PR TITLE
Add log_prob

### PR DESCRIPTION
**Pull request by masa-su**
* remove the `log_likelihood` function
* add `log_prob` in Distribution API which performs as the instance of Loss API.
* rename the `LossExpectation` class to `Expectation` and add the `expectation` method in Loss API.
* rename the `estimation` method in Loss API to `eval` (according to `model.eval()` on PyTorch).
* remove flow implementations for now.

### CODE REVIEW

**Comment by masa-su** [comment]
Example codes
```python
>> print(q)
Distribution:
  q(z|x) (Normal)
Network architecture:
  Inference(
    (fc1): Linear(in_features=784, out_features=512, bias=True)
    (fc2): Linear(in_features=512, out_features=512, bias=True)
    (fc31): Linear(in_features=512, out_features=64, bias=True)
    (fc32): Linear(in_features=512, out_features=64, bias=True)
  )

>> print(q.log_prob())
log q(z|x)  # log-probability density function（Loss API）

>> samples_dict = q.sample({"x": _x})
>> loss = q.log_prob().mean().estimate(samples_dict)
>> print(loss)
tensor(-67.4651, device='cuda:0', grad_fn=<MeanBackward1>)

>> loss = Expectation(q, (p * prior).log_prob() - q.log_prob()).mean()
>> print(loss)
mean(E_q(z|x)[log p(x,z) - log q(z|x)])

```
